site/en/docs/web-platform/navigation-api/index.md

```diff
@@ -39,32 +40,47 @@ In most cases, it lets your code override the browser's default behavior for tha
 For SPAs, that likely means keeping the user on the same page and loading or changing the site's content.
 
 A `NavigateEvent` is passed to the "navigate" listener which contains information about the navigation, such as the destination URL, and allows you to respond to the navigation in one centralized place.
-A basic "navigate" listener on example.com could look like this:
+A basic "navigate" listener could look like this:
 
 ```js
 navigation.addEventListener('navigate', navigateEvent => {
-  switch (navigateEvent.destination.url) {
-    case 'https://example.com/':
-      navigateEvent.transitionWhile(loadIndexPage());
-      break;
-    case 'https://example.com/cats':
-      navigateEvent.transitionWhile(loadCatsPage());
-      break;
+  const url = new URL(navigateEvent.destination.url);
```

Comment:
Maybe move this below the early bail-out.

Reply:
hah, yeah. I was passing it into `shouldNotIntercept` in an earlier draft.